src/main/scala/org/broadinstitute/hail/expr/Parser.scala

```diff
@@ -304,21 +295,47 @@ object Parser extends JavaTokenParsers {
 
   def structDeclaration: Parser[Array[(String, AST)]] = "{" ~> repsep(structField, ",") <~ "}" ^^ (_.toArray)
 
-  def structField: Parser[(String, AST)] = (stringLiteral ~ ":" ~ expr) ^^ { case id ~ _ ~ ast =>
-    (id.substring(1, id.length - 1), ast)
-  }
+  def structField: Parser[(String, AST)] = (stringLiteral ~ ":" ~ expr) ^^ { case id ~ _ ~ ast => (id, ast) }
 
   def indexStruct: Parser[(String, AST)] = "index" ~ "(" ~ expr ~ "," ~ identifier ~ ")" ^^ {
     case (_ ~ _ ~ ast ~ _ ~ key ~ _) => (key, ast)
     //    case (_ ~ _ ~ ast ~ _ ~ key ~ _) => (key.substring(1, key.length - 1), ast)
   }
 
+  def backtickLiteral: Parser[String] =
+    ("`" + """([^\\`]|\\[\\bfnrt`]|\\u[a-fA-F0-9]{4})*""" + "`").r ^^
+      (s => unescapeString(s.substring(1, s.length - 1))) |
```

Comment:
You don't lose it, you just have to compute it from the string by counting newlines and the position from the last newline.

Or do you mean lose it downstream, if we want to generate messages internal to the string?  That's true.  Once we unescape, we don't know which newlines were real and which were `\n`.  Although I don't think we do that anywhere.